src/bosh-softlayer-cpi/softlayer/common/vm_utils_test.go

```diff
@@ -704,5 +705,12 @@ var _ = Describe("VM Utils", func() {
 				Ω(fileContent).Should(ContainSubstring(appendDNSEntry))
 			})
 		})
+		Context("when the target file cannot be created", func() {
```

Comment:
Actually, I would expect a test for each error paths for this code snippet:
```
length, err := fmt.Fprintln(writer, "\n"+record)
if err != nil {
	return bosherr.WrapErrorf(err, "Writing '%s' to Writer", record)
}
if length != len(record)+2 {
	return bosherr.Errorf("The number (%d) of bytes written in Writer is not equal to the length (%d) of string", length, len(record)+2)
}
err = writer.Flush()
if err != nil {
  return bosherr.WrapErrorf(err, "Writing '%s' to file %s", record, path)
}
```
One for each error checks:
1. when `fmt.Fprintln` failed to write
2. when `length` written is not what expected (`length != len(record)+2`)
3. when `writer.Flush()`

Each are different error conditions that are all possible. When you test drive the code these tests are natural. Please add and I will review quickly and merge.

You are almost there.